backend/nomad_connection.go

```diff
@@ -641,18 +641,25 @@ func (c *NomadConnection) watchClientStats(action Action) {
 	}
 }
 
+func nodeUrl(params map[string]interface{}) string {
```

Comment:
I have no idea. I don't feel I'm being less defensive than the existing code; I'd have added checks if that was the style.